frontend/src/scenes/web-analytics/tabs/marketing-analytics/frontend/components/settings/AddSourceDropdown.tsx

```diff
@@ -0,0 +1,40 @@
+import { IconPlus } from '@posthog/icons'
+import { LemonButton, LemonDropdown } from '@posthog/lemon-ui'
+import { DataWarehouseSourceIcon } from 'scenes/data-warehouse/settings/DataWarehouseSourceIcon'
+
+import { ExternalDataSource, ManualLinkSourceType } from '~/types'
+
+interface AddSourceDropdownProps {
+    sources: ExternalDataSource['source_type'][] | ManualLinkSourceType[]
+    onSourceAdd: (source: any) => void // any because different files have different handler types
```

Comment:
I would still suggest having a parent type /interface here rather than just defaulting to a `any`. It could even help with the other parts where you have `required_fields_for_marketing_analytics`.

No need to do this on this PR, though. You can refactor this when you have some more sources to define the base interface.